consensus/istanbul/config.go

```diff
@@ -16,29 +16,99 @@
 
 package istanbul
 
-import "math/big"
+import (
+	"math/big"
 
-type ProposerPolicy uint64
+	"github.com/naoina/toml"
+)
+
+type ProposerPolicyId uint64
 
 const (
-	RoundRobin ProposerPolicy = iota
+	RoundRobin ProposerPolicyId = iota
 	Sticky
 )
 
+// ProposerPolicy represents the Validator Proposer Policy
+type ProposerPolicy struct {
+	Id       ProposerPolicyId    // Could be RoundRobin or Sticky
+	Registry []ValidatorSet      // Holds the ValidatorSet for a given block height
+	By       validatorSortByFunc // func that defines how the ValidatorSet should be sorted
+}
+
+// NewRoundRobinProposerPolicy returns a RoundRobin ProposerPolicy with ValidatorSortByStringFunc as default sort function
```

Comment:
what are the rules to do one or another?

Reply:
There are two types of validator sorting, sort by string and sort by byte. Legacy IBFT uses sort by string, where as QBFT uses sort by byte. When QBFT consensus is started, the sorting is set to sort by byte